website/content/docs/platform/k8s/vso/sources/vault.mdx

```diff
@@ -0,0 +1,283 @@
+during drift remediation.
+
+### Supported Vault authentication methods
+
+| Backend                                   | Description                                                                                                 |
+|-------------------------------------------|-------------------------------------------------------------------------------------------------------------|
+| [Kubernetes](/vault/docs/auth/kubernetes) | Relies on short-lived Kubernetes ServiceAccount tokens for Vault authentication                             |
+| [JWT](/vault/docs/auth/jwt)               | Relies on either static JWT tokens or short-lived Kubernetes ServiceAccount tokens for Vault authentication |
+| [AppRole](/vault/docs/auth/approle)       | Relies on static AppRole credentials for Vault authentication                                               |
+| [AWS](/vault/docs/auth/aws)               | Relies on AWS credentials for Vault authentication                                                          |
+
+## Vault access and custom resource definitions
+
+The Vault connection and authentication configuration is provided by the `VaultConnection` and `VaultAuth` CRDs. These can be considered as
+foundational Custom Resources that all secret replication type resources will reference.
+
+### VaultConnection custom resource
+
+Provides the configuration necessary for the Operator to connect to a single Vault server instance.
+
+```yaml
+---
+apiVersion: secrets.hashicorp.com/v1beta1
+kind: VaultConnection
+metadata:
+  namespace: vso-example
+  name: vault-connection
+spec:
+  # required configuration
+  # address to the Vault server.
+  address: http://vault.vault.svc.cluster.local:8200
+
+  # optional configuration
+  # HTTP headers to be included in all Vault requests.
+  # headers: []
+  # TLS server name to use as the SNI host for TLS connections.
+  # tlsServerName: ""
+  # skip TLS verification for TLS connections to Vault.
+  # skipTLSVerify: false
+  # the trusted PEM encoded CA certificate chain stored in a Kubernetes Secret
+  # caCertSecretRef: ""
+```
+
+### VaultAuth custom resource
+
+Provide the configuration necessary for the Operator to authenticate to a single Vault server instance as
+specified in a `VaultConnection` Custom Resource.
+
+```yaml
+---
+apiVersion: secrets.hashicorp.com/v1beta1
+kind: VaultAuth
+metadata:
+  namespace: vso-example
+  name: vault-auth
+spec:
+  # required configuration
+  # VaultConnectionRef of the corresponding VaultConnection CustomResource.
+  # If no value is specified the Operator will default to the `default` VaultConnection,
+  # configured in its own Kubernetes namespace.
+  vaultConnectionRef: vault-connection
+  # Method to use when authenticating to Vault.
+  method: kubernetes
+  # Mount to use when authenticating to auth method.
+  mount: kubernetes
+  # Kubernetes specific auth configuration, requires that the Method be set to kubernetes.
```

Comment:
I think it would be helpful to add the field `allowedNamespaces` here in a comment, or otherwise point out the feature since it's new and a fairly large impact on UX.

Reply:
True, I do link out to the api-reference, which includes more details on that feature. Would that be good enough?

Reply:
Yeah I think that is good enough for now.
I'll create an issue for me to add it to docs a bit more visibly when I get time because I think some chunk of users will want to use it.

Reply:
Good call. I think it would be good to provide more explicit docs around the way VSO with dependent resource references.